lib/src/lints/named_parameters_ordering/named_parameters_ordering_rule.dart

```diff
@@ -0,0 +1,180 @@
+/// conventions.
+///
+/// ### Configuration format
+/// ```yaml
+///  - named_parameters_ordering:
+///    order:
+///      - (parameterType)
+/// ```
+/// Where parameterType can be one of:
+///
+/// - `super`
+/// - `required_super`
+/// - `required`
+/// - `nullable`
+/// - `default`
+///
+/// ### Example:
+///
+/// Assuming config:
+///
+/// ```yaml
+/// custom_lint:
+///   rules:
+///     - named_parameters_ordering:
+///       order:
+///         - required
+///         - required_super
+///         - default
+///         - nullable
+///         - super
+/// ```
+///
+/// #### BAD:
+///
+/// ```dart
+/// class User {
+///   final String accountType;
+///   final String? userId;
+///
+///   User({
+///     required this.accountType,
+///     this.userId,
+///   });
+/// }
+///
+/// class UserProfile extends User {
+///   final String? age;
+///   final String? country;
+///   final String email;
+///   final bool isActive;
+///   final String name;
+///
+///   UserProfile({
+///     this.age,
+///     required super.accountType, // LINT, required super named parameters should be before nullable named parameters
+///     required this.name, // LINT, required named parameters should be before super named parameters
+///     super.userId,
+///     this.country, // LINT, nullable named parameters should be before super named parameters
+///     this.isActive = true, // LINT, default named parameters should be before nullable named parameters
+///     required this.email, // LINT, required named parameters should be before default named parameters
+///   });
+///
+///   void doSomething({
+///     required String name,
+///     int? age,
+///     bool isActive = true, // LINT, default named parameters should be before nullable named parameters
+///     required String email, // LINT, required named parameters should be before default named parameters
+///   }) {
+///     return;
+///   }
+/// }
+/// ```
+///
+/// #### GOOD:
+///
+/// ```dart
+/// class User {
+///   final String accountType;
+///   final String? userId;
+///
+///   User({
+///     required this.accountType,
+///     this.userId,
+///   });
+/// }
+///
+/// class UserProfile extends User {
+///   final String? age;
+///   final String? country;
+///   final String email;
+///   final bool isActive;
+///   final String name;
+///
+///   UserProfile({
+///     required this.name,
+///     required this.email,
+///     required super.accountType,
+///     this.isActive = true,
+///     this.age,
+///     this.country,
+///     super.userId,
+///   });
+///
+///   void doSomething({
+///     required String name,
+///     required String email,
+///     bool isActive = true,
+///     int? age,
+///   }) {
+///     return;
+///   }
+/// }
+/// ```
+class NamedParametersOrderingRule
+    extends SolidLintRule<NamedParametersOrderingParameters> {
+  /// This lint rule represents
+  /// the error whether we use bad formatted double literals.
```

Comment:
In the documentation for properties, you should explain the property, not the class 
```suggestion
  /// The name of this lint rule.
```